LXQt 0.15.0 and Friends

Chiitoo opened the pull request:
Not yet fully tested (so maybe no merge yet), but nitpicks are already very welcome.

The notable changes to previous ebuilds that I haven't forgotten about yet include:

- New package: lxqt-archiver
- Added a secondary SRC_URI, GitHub, mainly due to lxqt.org being down for the time being.  Additionally, at least some of the developers seem to consider it to be the primary source.
- Adjusted some licenses to have more of those mentioned in the sources.
- Dependency on 'polkit' is now optional.
- Some descriptions got updated/adjusted.

juippis commented on lxqt-base/lxqt-session/lxqt-session-0.15.0.ebuild:
```diff
@@ -0,0 +1,64 @@
+	dev-qt/qtx11extras:5
+	kde-frameworks/kwindowsystem:5[X]
+	=lxqt-base/liblxqt-${MY_PV}
+	x11-libs/libX11
+	x11-misc/xdg-user-dirs
+	themes? ( =x11-themes/lxqt-themes-${MY_PV} )
```
(Is this really needed in DEPEND? And a pure RDEPEND IUSE is banned by QA)
((I wouldn't care though since it's just for themes))

Chiitoo replied:
It is listed before 'session' in upstream directions here: https://github.com/lxqt/lxqt/wiki/Building-from-source#generalbuild-order

I'll see if I can test it out (I forget if I tested it out in the past already).

Chiitoo replied:
Yeah, doesn't seem to need to be there at build time at all, or I can't see a reason for it.  I actually don't know why the themes got tied to session in the first place.

Hm.  I guess they could be an unconditional dependency on the meta package or something?  I'm sure there will be someone who doesn't want them being installed...

Chiitoo replied:
Oh, the lxqt-meta package already does that... heh.

juippis commented on lxqt-base/lxqt-session/lxqt-session-0.15.0.ebuild:
```diff
@@ -0,0 +1,64 @@
+	cmake_src_install
+	doman lxqt-config-session/man/*.1 lxqt-session/man/*.1
+
+	echo XDG_CONFIG_DIRS=\"${EPREFIX}/usr/share\" >> 91lxqt-config-dir
+	doenvd 91lxqt-config-dir
```
`newend` here too, since ebuild helpers are more solid-proof than external commands.

Chiitoo replied:
Done.  Did I get it right?  Never dealt with these before.

juippis commented on x11-libs/libfm-qt/libfm-qt-0.15.0.ebuild:
```diff
@@ -0,0 +1,50 @@
+	dev-qt/qtwidgets:5
+	dev-qt/qtx11extras:5
+	>=lxde-base/menu-cache-1.1.0:=
+	media-libs/libexif:=
```
This doesn't seem to have a subslot, so the `:=` here seems useless.

Chiitoo replied:
I was actually wondering about that too.  I looked at the devmanual, and it says that "If an ebuild does not explicitly declare a sub-slot, the regular slot is used as the value of the sub-slot by default.", so I figured it was put there to force a rebuild in any case.

(Did that package maybe have a subslot in the past?)

During all this I noticed that some of the LXQt subslots had not been updated since 0.13 maybe... ooops.

juippis commented on x11-misc/pcmanfm-qt/pcmanfm-qt-0.15.0.ebuild:
```diff
@@ -0,0 +1,59 @@
+pkg_postinst() {
+	xdg_desktop_database_update
+
+	if ! has_version lxqt-base/lxqt-meta && ! has_version gnome-base/gvfs; then
+		elog
+		elog "To make use of the 'trash' functionality, please install"
+		elog "the 'gnome-base/gvfs' package."
+		elog
+	fi
```
`optfeature` from *eutils.eclass* will only be printed if the said package is not installed.

But these are taste preferences.

(to clarify, I'd do 
```
if ! has_version lxqt-base/lxqt-meta && optfeature "trash functionality" gnome-base/gvfs
```
or something like that)

Chiitoo replied:
I've meant to look into it, as I've seen it in some other ebuilds.  I'll see if I can do so for this.

juippis commented on x11-misc/qps/qps-2.1.0.ebuild:
```diff
@@ -0,0 +1,41 @@
+LICENSE="GPL-2 GPL-2+ LGPL-2.1+ QPL-1.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+IUSE="debug"
```
This doesn't seem to be doing anything?

... unless it somehow interacts with *cmake.eclass* that I'm not aware of?

a17r replied:
This previously made cmake.eclass add -DNDEBUG flag but it was dropped in commit 95577dd5076a8e9864e82879fd3af97cf63fcfe9

Chiitoo replied:
Will look into this one as well.

Chiitoo replied:
Seems like the only reference I find, '#ifndef NDEBUG', in the sources is commented out, so I guess I'll drop the 'IUSE' from this one.

juippis commented on x11-themes/lxqt-themes/lxqt-themes-0.15.0.ebuild:
```diff
@@ -0,0 +1,27 @@
+SLOT="0"
+
+BDEPEND=">=dev-util/lxqt-build-tools-0.7.0"
+DEPEND=""
+RDEPEND="${DEPEND}"
```
These seem rather useless.

Chiitoo replied:
They shall be destroyed.

a17r commented on lxqt-base/lxqt-meta/lxqt-meta-0.15.0.ebuild:
```diff
@@ -0,0 +1,58 @@
+LICENSE="metapackage"
+SLOT="0"
+
+IUSE="+about admin archiver +filemanager lightdm lximage minimal
+	nls	+policykit powermanagement processviewer screenshot sddm
```
I know you have been thinking about lxqt-meta use flags lately, this is just a suggestion taken from plasma-meta: `+display-manager` if you think having a display manager by default would make sense; and then only `sddm` to switch between that and lightdm. Preference on whatever upstream may recommend or you think fits best.

Chiitoo replied:
Yeah, seems like it would make sense.  Will likely try it out.

Chiitoo commented on lxqt-base/lxqt-meta/lxqt-meta-0.15.0.ebuild:
```diff
@@ -0,0 +1,58 @@
+SLOT="0"
+
+IUSE="+about admin archiver +filemanager lightdm lximage minimal
+	nls	+policykit powermanagement processviewer screenshot sddm
```
Wow, I even knew about it, and thought I removed it.  It shows up as a single space in nano.  What.  Yes I use nano, so sue me!  (Don't actually... pretty please.  And to be fair(?), it's a single space there in many other editors too, but I digress!)

Don't know how I missed it in 'git diff'.  Many thanks for spotting it!

Chiitoo commented on lxqt-base/lxqt-openssh-askpass/lxqt-openssh-askpass-0.15.0.ebuild:
```diff
@@ -0,0 +1,45 @@
+
+	echo "SSH_ASKPASS='${EPREFIX}/usr/bin/lxqt-openssh-askpass'" >> "${T}/99${PN}" \
+		|| die
+	doenvd "${T}/99${PN}"
```
Done.

Chiitoo commented:
> What a massive PR.

It is sizeable, huh.  I might do any future ones in smaller chunks, where possible...

Updates:

- Builds look good (didn't try every possible combination, but did try enabling all the ones on lxqt-meta and lxqt-panel, as well as some others, which pull most of the dependencies in.
- Fix for bug 720716 [1].
- Actually remove the unconditional 'sys-auth/polkit-qt' where 'backlight? ( sys-auth/polkit-qt )' was added in 'lxqt-base/liblxqt'... not sure how I managed to leave it there.
- Added USE="display-manager" to 'lxqt-base/lxqt-meta'.
- Backported two changes to 'x11-misc/pcmanfm-qt',  Point release will come in a couple of weeks.
- Added bump for 'x11-misc/obconf-qt' which I forgot about previously.
- Anything I'm forgetting right now...  I need to start making notes.
- Plus all the earlier comments should be addressed.

Possible TODO: The USE="minimal" is a bit weird in 'lxqt-base/lxqt-meta'.  Pondering on something like 'window-manager', similar to 'display-manager'.

Chiitoo commented:
May as well wait for at least one more release of lxqt-archiver, which will have a GCC-10 build fix in.

Chiitoo commented:
- Changed HOMEPAGE to GitHub since 'lxqt.org' is still down, and its future is uncertain.  Upstream considers GitHub the main resource at this time.
- Bumped lxqt-archiver to 0.2.0, pcmanfm-qt to 0.15.1, and libfm-qt to 0.15.1.
- Updated date on masks due to a conflict.

Chiitoo commented:
Builds look good still/again, so I think I'll call this one done, unless someone spots something new still.

Thanks for all the reviewing!

Chiitoo commented:
Backported fix for 1391 [1],

1. https://github.com/lxqt/lxqt-panel/issues/1391

Chiitoo commented:
One more point release joins the party (lxqt-panel-0.15.1).
